src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -187,6 +197,9 @@ public function loadClass($class)
                     if ($parent && isset(self::$deprecated[$parent]) && strncmp($ns, $parent, $len)) {
                         @trigger_error(sprintf('The %s class extends %s that is deprecated %s', $name, $parent, self::$deprecated[$parent]), E_USER_DEPRECATED);
                     }
+                    if ($parent && isset(self::$final[$parent]) && strncmp($ns, $parent, $len)) {
+                        @trigger_error(sprintf('Extending the class %s in %s is deprecated and may break in the next major release.', $parent, $name), E_USER_DEPRECATED);
```

Comment:
It doesn't contain the versions in which the deprecation was added and when it will be applied compared to deprecation messages but maybe that's enough.